client/js/app/components/explorer/query_builder/group_by_field.js

```diff
@@ -24,25 +24,79 @@ var GroupByField = React.createClass({
     return this.props.value;
   },
 
+  handleChange: function(name, value) {
+    var newVal = this.props.value.slice();
+    var index = name.split('.')[1];
+    newVal[index] = value;
+    this.props.handleChange(name.split('.')[0], newVal);
+  },
+
+  toggleMultiGroupBy: function(event) {
+    event.preventDefault();
+    var newVal;
+    switch (this.props.value.length) {
+      case 0:
```

Comment:
The array is empty at first when no `group_by` has been set yet. So initial state basically.